app/com/linkedin/drelephant/tuning/FitnessComputeUtil.java

```diff
@@ -70,52 +89,64 @@ public FitnessComputeUtil() {
    */
   public void updateFitness() {
     logger.info("Computing and updating fitness for completed executions");
-    List<TuningJobExecution> completedExecutions = getCompletedExecutions();
-    updateExecutionMetrics(completedExecutions);
-    updateMetrics(completedExecutions);
+    List<TuningJobExecutionParamSet> completedJobExecutionParamSets = getCompletedJobExecutionParamSets();
+    updateExecutionMetrics(completedJobExecutionParamSets);
+    updateMetrics(completedJobExecutionParamSets);
 
     Set<JobDefinition> jobDefinitionSet = new HashSet<JobDefinition>();
-    for (TuningJobExecution tuningJobExecution : completedExecutions) {
-      jobDefinitionSet.add(tuningJobExecution.jobExecution.job);
+    for (TuningJobExecutionParamSet completedJobExecutionParamSet : completedJobExecutionParamSets) {
+      JobDefinition jobDefinition = completedJobExecutionParamSet.jobSuggestedParamSet.jobDefinition;
+      if (isTuningEnabled(jobDefinition.id)) {
+        jobDefinitionSet.add(jobDefinition);
+      }
     }
     checkToDisableTuning(jobDefinitionSet);
   }
 
   /**
    * Checks if the tuning parameters converge
-   * @param jobExecutions List of previous executions on which parameter convergence is to be checked
+   * @param tuningJobExecutionParamSets List of previous executions and corresponding param sets
    * @return true if the parameters converge, else false
    */
-  private boolean doesParameterSetConverge(List<JobExecution> jobExecutions) {
+  private boolean didParameterSetConverge(List<TuningJobExecutionParamSet> tuningJobExecutionParamSets) {
     boolean result = false;
     int num_param_set_for_convergence = 3;
```

Comment:
Can you change this to camelCase? Probably leaked as part of some other PR.

Can you also update num_fitness_for_median variable below?

Reply:
Done. I am wondering why I didn't use camelCase! Thanks for pointing out.